acs-apiclient/iOS/ExternalFlowWebViewController.cs

```diff
@@ -0,0 +1,168 @@
+        /// </summary>
+        void PrepareView()
+        {
+            View.BackgroundColor = UIColor.White;
+
+            PrepareNavigationBar();
+            PrepareBackButton();
+            PrepareRefreshButton();
+            PrepareNavigationItem();
+            PrepareWebView();
+        }
+
+        void PrepareNavigationBar()
+        {
+            navigationBar.TintColor = UIColor.Black;
+            navigationBar.BackgroundColor = UIColor.White;
+            navigationBar.Translucent = false;
+        }
+
+        void PrepareBackButton()
+        {
+            var closeIcon = UIImage.FromBundle("ic_close_36pt");
+            dismissButton = new UIBarButtonItem(closeIcon, UIBarButtonItemStyle.Plain, (sender, e) =>
+            {
+                webViewLoginDelegate.DidCancelLoginProcess();
+                DismissViewController(true, null);
+            });
+        }
+
+        void PrepareRefreshButton()
+        {
+            refreshButton = new UIBarButtonItem(UIBarButtonSystemItem.Refresh);
+            refreshButton.Clicked += (sender, e) =>
+            {
+                webView.LoadRequest(new NSUrlRequest(urlToLoad));
+            };
+        }
+
+        void PrepareNavigationItem()
+        {
+            NavigationItem.Title = urlToLoad.Host;
+            NavigationItem.LeftBarButtonItem = dismissButton;
+            NavigationItem.RightBarButtonItem = refreshButton;
+        }
+
+        void PrepareWebView()
+        {
+            webView = new WKWebView(CGRect.Empty, new WKWebViewConfiguration());
+            webView.AllowsBackForwardNavigationGestures = false;
+            webView.BackgroundColor = UIColor.White;
+            webView.TranslatesAutoresizingMaskIntoConstraints = false;
+            webView.NavigationDelegate = this;
+
+            View.AddSubview(webView);
+
+            webView.TopAnchor.ConstraintEqualTo(View.TopAnchor).Active = true;
+            webView.LeftAnchor.ConstraintEqualTo(View.LeftAnchor).Active = true;
+            webView.BottomAnchor.ConstraintEqualTo(View.BottomAnchor).Active = true;
+            webView.RightAnchor.ConstraintEqualTo(View.RightAnchor).Active = true;
+
+            webView.LayoutIfNeeded();
+        }
+
+        #endregion
+    }
+
+    partial class ExternalFlowWebViewController: IWKNavigationDelegate
+    {
+        /// <summary>
+        /// Decides the policy to use for a new navigation action.
+        /// </summary>
+        /// <param name="webView">The web view which is requesting the navigation action.</param>
+        /// <param name="navigationAction">The navigation action.</param>
+        /// <param name="decisionHandler">The decision callback.</param>
+        [Export("webView:decidePolicyForNavigationAction:decisionHandler:")]
+        public void DecidePolicy(
+            WKWebView webView,
+            WKNavigationAction navigationAction,
+            Action<WKNavigationActionPolicy> decisionHandler
+        ) {

```

Comment:
:bowtie: We need to come unp with  convention here. I'm definitely more of a fan of keeping brackets on the same line as the final argument:
```
        public void DecidePolicy(
            WKWebView webView,
            WKNavigationAction navigationAction,
            Action<WKNavigationActionPolicy> decisionHandler) 
        {
            try
```

But also willing to bow to C# conventions or you guys' preference...

Reply:
✅